src/main/java/spoon/support/reflect/declaration/CtRecordComponentImpl.java

```diff
@@ -36,18 +39,45 @@ public class CtRecordComponentImpl extends CtNamedElementImpl implements CtRecor
 	public CtMethod<?> toMethod() {
 		CtMethod<?> method = this.getFactory().createMethod();
 		method.setSimpleName(getSimpleName());
-		method.setType((CtTypeReference) getType());
+		method.setType(getType() != null ? getType().clone() : null);
 		method.setExtendedModifiers(Collections.singleton(new CtExtendedModifier(ModifierKind.PUBLIC, true)));
 		method.setImplicit(true);
-		method.setBody(getFactory().createCodeSnippetStatement("return " + getSimpleName()));
+
+		CtFieldAccess<?> ctVariableAccess = (CtFieldAccess<?>) getFactory().Code()
+			.createVariableRead(getRecordFieldReference(), false);
+		// ensure the `this.` is implicit (important as we are also implicit)
```

Comment:
why is this important? the variable access and the return are not implicit

Reply:
I can't remember what the reason for this comment was, the PR has been open for months. I guess this could be removed

Reply:
I would imagine you wanted `return foo` instead of `return this.foo`, as the whole synthetic method should arguably be implicit (it does not exist in the source). WDYT @SirYwell? Do we want the whole thing to be implicit (method and all) or explicit?